inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/chart/ChartPanel.java

```diff
@@ -0,0 +1,189 @@
+import de.tudarmstadt.ukp.inception.recommendation.model.LearningCurve;
+
+public class ChartPanel
+    extends Panel
+{
+    private static final long serialVersionUID = -3849226240909011148L;
+
+    private static final Logger LOG = LoggerFactory.getLogger(ChartPanel.class);
+
+    private static final String MID_CHART_CONTAINER = "chart";
+
+    private LoadableDetachableModel<LearningCurve> model;
+    private final WebMarkupContainer chart;
+    private final ChartAjaxBejavior chartAjaxBejavior;
+
+    public ChartPanel(String aId, LoadableDetachableModel<LearningCurve> aModel)
+    {
+        super(aId, aModel);
+        model = aModel;
+        
+        chart = new WebMarkupContainer(MID_CHART_CONTAINER);
+        chart.setMarkupId("canvas");
+        chart.setOutputMarkupId(true);
+        add(chart);
+
+        chartAjaxBejavior = new ChartAjaxBejavior();
+        add(chartAjaxBejavior);
+    }
+    
+    @Override
+    public void renderHead(IHeaderResponse aResponse)
+    {
+        super.renderHead(aResponse);
+        
+        // import Js
+        aResponse.render(JavaScriptHeaderItem
+                .forReference(new WebjarsJavaScriptResourceReference("c3/current/c3.js")));
+        aResponse.render(JavaScriptHeaderItem
+                .forReference(new WebjarsJavaScriptResourceReference("d3js/current/d3.js")));
+
+        // import Css
+        aResponse.render(
+                CssHeaderItem.forReference(new WebjarsCssResourceReference("c3/current/c3.css")));
+
+        aResponse.render(JavaScriptReferenceHeaderItem.forReference(
+                getApplication().getJavaScriptLibrarySettings().getJQueryReference()));
+        
+        aResponse.render(JavaScriptHeaderItem.forReference(ChartJsReference.get()));
+
+        String chartTriggerJavascript = String.join("\n",
+                "$(document).ready(function(){", 
+                "   $.ajax({",
+                "       url:'" + chartAjaxBejavior.getCallbackUrl().toString() + "',",
+                "       type:'post',",
+                "       cache:!1,",
+                "       contentType:'application/json',",
+                "       dataType:'json',",
+                "       success : function(result){",
+                "           updateLearningCurveDiagram(result)",
+                "       }",
+                "   })",
+                "})");
+
+        aResponse.render(JavaScriptContentHeaderItem.forScript(chartTriggerJavascript, null));
+    }
+ 
+    @Override
+    protected void onBeforeRender()
+    {
+        chart.add(new AttributeModifier("my:canvas.chartid", getMarkupId()));
+        super.onBeforeRender();
+    }
+
+    private final class ChartAjaxBejavior
+        extends AbstractAjaxBehavior
+    {
+        private static final long serialVersionUID = 1L;
+
+        @Override
+        protected void onComponentTag(final ComponentTag tag)
+        {
+            tag.put("my:canvas.chartid", getMarkupId());
```

Comment:
What does this `my:canvas.chartid` do?

Reply:
Used it to create an attribute in the chart div where the chart is being drawn by the JS. I think this can also be done by using id only.

Reply:
If you call `setOutputMarkupId(true)` on a component, then Wicket automatically generates an `id` attribute and you can fetch the value of that attribute using `getMarkupId()` - so there shouldn't be any need to stuff the ID in a custom attribute.